retriever/lib/engine_tools.py

```diff
@@ -151,7 +151,9 @@ def walker(dictionary, row_key=None, header_values=None, rows=[], normalize=Fals
             return rows
 
     if isinstance(dictionary, dict):
-        if header_values and (header_values in list(dictionary.keys()) or header_values == list(dictionary.keys())):
+        print(header_values, dictionary.keys())
+        print()
+        if header_values and (set(dictionary.keys()).issubset(header_values) or header_values == list(dictionary.keys())):
```

Comment:
I guess you want to do `set(header_values).issubset(dictionary.keys())`. Also, after this `header_values == list(dictionary.keys())` may not be required.